### PR TITLE
Adding material to compute the heat transfer coefficient

### LINKED ISSUE #23709: Add a material to calculate heat transfer coefficient with Wolf-McCarthy correlation
## Reason

The Wolf-McCarthy correlation is commonly used for hydrogen flow
## Design

Implement a material.
## Impact

Add a new option to calculate HTCs instead of coding them in the input file

### PR DESCRIPTION
**Pull request by licharlot**
Using the Wolf-McCarthy correlation.

Closes #23709

### CODE REVIEW

File: modules/thermal_hydraulics/doc/content/source/materials/ADWallHeatTransferCoefficientWolfMcCarthyMaterial.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# ADWallHeatTransferCoefficientWolfMcCarthyMaterial
+
+The material computes the convective heat transfer coefficient using the Wolf-McCarthy correlation  (see [!cite](wolf_mccarthy1960)).
+
+The Nusselt number is calculated as:
+
+\begin{equation}
+  Nu= 0.025 Re^{0.8} Pr^{0.4} \left(\frac{T_{fluid}}{T_{wall}}\right)^{-0.55}
```
Comment (veeshy) [inline modules/thermal_hydraulics/doc/content/source/materials/ADWallHeatTransferCoefficientWolfMcCarthyMaterial.md:8]: I think you shouldn’t have the minus sign in the exponent of the bulk to wall. Any chance you could give an eq# or page number for the ref?

Comment (licharlot) [reply modules/thermal_hydraulics/doc/content/source/materials/ADWallHeatTransferCoefficientWolfMcCarthyMaterial.md:8]: I am using Equation 25 in this report. https://www.osti.gov/servlets/purl/4842792 . There are several forms in this report, but I think this is the one that is most commonly used.

Comment (veeshy) [reply modules/thermal_hydraulics/doc/content/source/materials/ADWallHeatTransferCoefficientWolfMcCarthyMaterial.md:8]: It’s a typo in the report? Wall over bulk is used everywhere else, even in figure 25 where the text says eq 25 is compared to experiments.

Comment (licharlot) [reply modules/thermal_hydraulics/doc/content/source/materials/ADWallHeatTransferCoefficientWolfMcCarthyMaterial.md:8]: You are right, this must be a typo in the report. I'll fix it. Thanks for the catch!

Comment (joshuahansel) [reply modules/thermal_hydraulics/doc/content/source/materials/ADWallHeatTransferCoefficientWolfMcCarthyMaterial.md]: Agreed, nice catch! I probably wouldn't have caught it in my review...

Lise, can you make a note in your doc where you reference that doc to mention that there is a typo in the reference?

File: modules/thermal_hydraulics/doc/content/source/materials/ADWallHeatTransferCoefficientWolfMcCarthyMaterial.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
+# ADWallHeatTransferCoefficientWolfMcCarthyMaterial
+
+The material computes the convective heat transfer coefficient using the Wolf-McCarthy correlation  (see [!cite](wolf_mccarthy1960)).
```
Comment (joshuahansel) [inline modules/thermal_hydraulics/doc/content/source/materials/ADWallHeatTransferCoefficientWolfMcCarthyMaterial.md:3]: Mention the typo in the report.

**Review by joshuahansel — CHANGES_REQUESTED** [review]

File: modules/thermal_hydraulics/include/materials/ADWallHeatTransferCoefficientWolfMcCarthyMaterial.h [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "Material.h"
+
+/**
+ * Computes wall heat transfer coefficient using Dittus-Boelter equation
```
Comment (GiudGiud) [inline modules/thermal_hydraulics/include/materials/ADWallHeatTransferCoefficientWolfMcCarthyMaterial.h:15]:
```suggestion
 * Computes wall heat transfer coefficient using the Wolf-McCarthy correlation
```

Comment (joshuahansel) [reply modules/thermal_hydraulics/include/materials/ADWallHeatTransferCoefficientWolfMcCarthyMaterial.h]: Done externally

**Review by joshuahansel — APPROVED** [review]

**Pull request merged**